src/Engine/ProtoCore/Namespace/SymbolTable.cs

```diff
@@ -103,6 +242,28 @@ public override bool Equals(object obj)
         }
 
         /// <summary>
+        /// Equality operator overload to have same equality behaviour as object.Equals() override
+        /// </summary>
+        /// <param name="thisSymbol"></param>
+        /// <param name="otherSymbol"></param>
+        /// <returns></returns>
+        public static bool operator ==(Symbol thisSymbol, Symbol otherSymbol)
```

Comment:
I think it's other way, when you are overloading operator ==, you must override object.Equals. If you have overridden object.Equals then even if you don't overload operator ==, it should call object.Equals internally.